pkg/controller/secrets/tenant_controller.go

```diff
@@ -90,6 +90,13 @@ func AddTenantController(mgr manager.Manager, opts options.AddOptions) error {
 		return fmt.Errorf("tenant-controller failed to watch tenant CA Secret %s in all namespace: %w", certificatemanagement.TenantCASecretName, err)
 	}
 
+	if err = utils.AddConfigMapWatch(c, certificatemanagement.TrustedCertConfigMapName, "", &handler.EnqueueRequestForObject{}); err != nil {
```

Comment:
First of all, you adding this statement twice.
Secondly, why would we want to watch the things that we create? What is there to gain by this?